### PR TITLE
added metrics for inference/InvokeAI

### PR DESCRIPTION
**Pull request by luca-brg**
Code to track the metrics

### CODE REVIEW

File: invokeai/backend/generate.py [hunk 1]
```diff
@@ -42,6 +42,19 @@
 from .util import choose_precision, choose_torch_device, image_to_base64, upload_on_blob
 from .prompting.prompt_template import standard_fashion_prompt
```
Comment (vaidehiResleeve) [inline invokeai/backend/generate.py:44]: maybe remove the space.

File: invokeai/backend/generate.py [hunk 2]
```diff
@@ -42,6 +42,19 @@
...
+request_count = Counter('generate_requests_total', 'Total number of generate requests')
+request_failed_count = Counter('generate_requests_failed_total', 'Total number of generate requests failed')
+generated_count = Counter("generated_count", "Total Number of successful images generated")
+img2img_count = Counter("img2img_requests_total", "Total number of img2img generate requests")
+txt2img_count = Counter("txt2img_requests_total", "Total number of txt2img generate requests")
+generation_time = Histogram("generation_time_seconds", "Total time taken to generate a final image (in seconds)")
+
+#monitor the metrics on port 9091
+start_http_server(9091)
+
```
Comment (vaidehiResleeve) [inline invokeai/backend/generate.py:57]: this space as well

File: invokeai/backend/generate.py [hunk 3]
```diff
@@ -610,15 +634,25 @@ def process_image(image,seed):
...
         except RuntimeError:
+            # Increase counter of failed requests
+            request_failed_count.inc()
             # Clear the CUDA cache on an exception
             self.clear_cuda_cache()
 
             print(traceback.format_exc(), file=sys.stderr)
             print(">> Could not generate image.")
 
+
```
Comment (vaidehiResleeve) [inline invokeai/backend/generate.py:645]: remove space here as well.

File: invokeai/backend/generate.py [hunk 4]
```diff
@@ -602,6 +623,9 @@ def process_image(image,seed):
             )
 
         except KeyboardInterrupt:
+            # We don't support anymore keyboard interruption.
```
Comment (vaidehiResleeve) [inline invokeai/backend/generate.py:626]: just a question : why is the significance of these two comments?

**Review by vaidehiResleeve — CHANGES_REQUESTED** [review]
Just few spaces here and there.

File: invokeai/backend/generate.py (hunk 4 above)
Comment (luca-brg) [reply invokeai/backend/generate.py:626]: It was to explain why we are not handling this error. But considering this part of code is never triggered anymore we could just consider removing it.

File: invokeai/backend/generate.py (hunk 1 above)
Comment (luca-brg) [reply invokeai/backend/generate.py:44]: Fixed it

File: invokeai/backend/generate.py (hunk 2 above)
Comment (luca-brg) [reply invokeai/backend/generate.py:57]: Done.

File: invokeai/backend/generate.py (hunk 3 above)
Comment (luca-brg) [reply invokeai/backend/generate.py:645]: And done.

File: invokeai/backend/generate.py (hunk 4 above)
Comment (vaidehiResleeve) [reply invokeai/backend/generate.py]: maybe we can keep this comment for now and later review if we want to implement or not.

**Review by vaidehiResleeve — APPROVED** [review]
approved

**Pull request merged**